pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package broker
+
+// this was copied from where else and edited to fit our objects
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+/* Begin Create Definition */
+
+type brokerCreateStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+	kapi.NameGenerator  // GenerateName method for CreateStrategy
+}
+
+// implements RESTCreateStrategy interface
+var createStrategy = brokerCreateStrategy{
+	// embeds to pull in existing code behavior from upstream
+
+	// this has an interesting NOTE on it. Not sure if it applies to us.
+	ObjectTyper: kapi.Scheme,
+	// use the generator from upstream k8s, or implement method
+	// `GenerateName(base string) string`
+	NameGenerator: kapi.SimpleNameGenerator,
+}
+
+// Canonicalize is called after validate. What happens if it creates
+// an object to persist that does not pass validate? (I think the
```

Comment:
I'm actually going through and enhancing comments on some of this stuff in k8s now.  Canonicalize is a chance to mutate the object into a canonical form so that code that operates on a resource can rely on a common form for operations like comparison.  One example is endpoints - which are canonicalized by packing into a canonical form and then sorted in this method.